src/ir/datalog/input_relation_fact_test.cc

```diff
@@ -70,11 +76,13 @@ class OneOfEachFact
 };
 
 TEST(OneOfEachFactTest, OneOfEachFactTest) {
-  EXPECT_EQ(OneOfEachFact(Number(5), Symbol("foo"),
-                          Record<Number, Symbol>(Number(3), Symbol("bar")),
-                          SimpleAdt(UnitAdtBranch()))
-                .ToDatalogString(),
+  auto fact = OneOfEachFact(Number(5), Symbol("foo"),
+                            Record<Number, Symbol>(Number(3), Symbol("bar")),
+                            SimpleAdt(UnitAdtBranch()));
+  EXPECT_EQ(fact.ToDatalogString(),
             R"(oneOfEach(5, "foo", [3, "bar"], $Unit()).)");
+  EXPECT_EQ(fact.ToDatalogFactsFileString(),
+            R"(5, "foo", [3, "bar"], $Unit())");
```

Comment:
Ah, records will also pose a problem with `,` being your separator. Note here that `,` is used to both separate the fields in the record and delimit the boundary between the column containing the record and the next and prior column. Souffle will get very confused if it tries to read this line from an input file.

Reply:
I replicated whatever was in `ToDatalogString` without the relation name. What is the right format?

Reply:
You can choose whatever delimiter you want to separate rows in Souffle. I tend to choose `;`, which would cause the expected output to be as follows:

```C++
 R"(5; "foo"; [3, "bar"]; $Unit())"
 ```